frontend/app-development/features/appContentLibrary/AppContentLibrary.test.tsx

```diff
@@ -7,25 +7,53 @@ import { createQueryClientMock } from 'app-shared/mocks/queryClientMock';
 import { QueryKey } from 'app-shared/types/QueryKey';
 import { app, org } from '@studio/testing/testids';
 import { queriesMock } from 'app-shared/mocks/queriesMock';
+import type { UserEvent } from '@testing-library/user-event';
 import userEvent from '@testing-library/user-event';
-import type { Option } from 'app-shared/types/Option';
+import type { OptionsLists } from 'app-shared/types/api/OptionsLists';
+import type { CodeList } from '@studio/components';
 
-const optionListsMock: Record<string, Option[]> = {
+const uploadCodeListButtonTextMock = 'Upload Code List';
+const updateCodeListButtonTextMock = 'Update Code List';
+const codeListNameMock = 'codeListNameMock';
+const codeListMock: CodeList = [{ value: '', label: '' }];
+jest.mock(
+  '../../../libs/studio-content-library/src/ContentLibrary/LibraryBody/pages/CodeList',
+  () => ({
+    CodeList: ({ onUpdateCodeList, onUploadCodeList }: any) => (
+      <div>
+        <button
+          onClick={() =>
+            onUploadCodeList(
+              new File(['test'], `${codeListNameMock}.json`, { type: 'application/json' }),
+            )
+          }
+        >
+          {uploadCodeListButtonTextMock}
+        </button>
+        <button
+          onClick={() => onUpdateCodeList({ title: codeListNameMock, codeList: codeListMock })}
+        >
+          {updateCodeListButtonTextMock}
+        </button>
+      </div>
+    ),
+  }),
+);
```

Comment:
What is preventing the actual CodeList component to be used here instead of a mock?

Reply:
Nothing is preventing it. But I figured that it is out of scope for this component in `app-development` to care about how the `CodeList` inside the library is implemented and interact with it only to trigger the callback that this component in `app-development` is interested in. 
So testing the functionality of the CodeList should be covered by the library and not this component 😄